iphone/Maps/Classes/MapsAppDelegate.mm

```diff
@@ -556,6 +554,9 @@ - (void)applicationWillResignActive:(UIApplication *)application
 {
   [self.mapViewController.appWallAd close];
   [RouteState save];
+  
+  [self.locationManager onBackground];
+  [self.mapViewController onEnterBackground];
```

Comment:
Именно так: если читать всё, то как раз и становится понятно, что OGL методы из didEnterBackground система еще позволяет. Напомню, что последние пять лет именно так у нас код и работал.

> On Mar 21, 2016, at 13:41, Илья Гречухин <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/MapsAppDelegate.mm <https://github.com/mapsme/omim/pull/2365#discussion_r56803810>:
> 
> > @@ -556,6 +554,9 @@ - (void)applicationWillResignActive:(UIApplication *)application
> >  {
> >    [self.mapViewController.appWallAd close];
> >    [RouteState save];
> > +  
> > +  [self.locationManager onBackground];
> > +  [self.mapViewController onEnterBackground];
> Ну уж если читаем внимательно, то читаем все:
> In your app delegate’s applicationWillResignActive: method, your app should stop its animation timer (if any), place itself into a known good state, and then call the glFinish function.
> 
> In your app delegate’s applicationDidEnterBackground: method, your app may want to delete some of its OpenGL ES objects to make memory and resources available to the foreground app. Call the glFinish function to ensure that the resources are removed immediately.
> 
> After your app exits its applicationDidEnterBackground: method, it must not make any new OpenGL ES calls. If it makes an OpenGL ES call, it is terminated by iOS.
> 
> In your app’s applicationWillEnterForeground: method, re-create any objects and restart your animation timer.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2365/files/4c1ad60019dfa841acbca6cd814eb2661ea98b14#r56803810>